src/lib.rs

```diff
@@ -42,7 +42,7 @@ pub mod http;
 #[cfg(all(feature = "std", esp_idf_comp_esp_http_server_enabled))]
 pub mod httpd;
 pub mod log;
-#[cfg(feature = "alloc")]
+#[cfg(all(feature = "alloc", esp_idf_comp_mdns_enabled))]
```

Comment:
let's leave the work of making sure that the `mdns` component is fetched to `esp-idf-sys`, no need to check `espidf` version here

Reply:
But do we have `esp_idf_comp_mdns_enabled` as a valid cfg in ESP-IDF V4.4? I think so, but if you could confirm. I don't want that we are ending with regressions for V4.4.

Reply:
`esp_idf_comp_mdns_enabled` is a valid cfg in v4.4